src/utils/abi-decoder.ts

```diff
@@ -219,11 +218,12 @@ export default class ABIDecoder {
                         try {
                             decodedData = getDecodedData(logItem, abiItem)
                         } catch (e) {
-                            //try again
+                            abiItemOptions = (await this.db.getEventABIsForHexSignature(eventID)) || []
                         }
-                    } else if (abiItemOptions.length > 0) {
+                    }
+                    if (abiItemOptions.length > 0) {
                         // try all of the options, it'll throw an error if it doesn't match, catch it, try the next one
-                        while (!decodedData && abiItemOptions.length > 0) {
+                        while ((!decodedData || !Object.keys(decodedData).length) && abiItemOptions.length > 0) {
```

Comment:
Do you remember what specific case we needed this for?

if there isn't any `logData`, this will actually be an empty object.

Thinking about it a bit more, I just have this requirement only be there if logData isn't blank